kubectl-cloudflow/deploy/deploy.go

```diff
@@ -379,3 +695,67 @@ func prefixWithStreamletName(streamletName string, key string) string {
 func formatStreamletConfigKeyFq(streamletName string, key string) string {
 	return fmt.Sprintf("cloudflow.streamlets.%s", prefixWithStreamletName(streamletName, key))
 }
+
+/*
+{
+  blueprint : {
+    name : taxi-ride-fare
+    images : {
+      proc : docker.io/lightbend/processor:1.0
+      ings : docker.io/lightbend/ingestor:1.0
+      logr : docker.io/lightbend/logger:1.0
+    }
+    streamlets : {
+      taxi-ride : ings/taxiride.ingestor.TaxiRideIngress
+      taxi-fare : ings/taxiride.ingestor.TaxiFareIngress
+      processor : proc/taxiride.processor.TaxiRideProcessor
+      logger : logr/taxiride.logger.FarePerRideLogger
+    }
+    connections : {
+      taxi-ride : {
+        out : [processor.in-taxiride]
+      }
+      taxi-fare : {
+        out : [processor.in-taxifare]
+      }
+      processor : {
+        out : [logger.in]
+      }
+    }
+  }
+}
+*/
+
+// ImageReference is a destructured version of the image path
+type ImageReference struct {
+	Registry   string
+	Repository string
+	Image      string
+	Tag        string
+	FullURI    string
+}
+
+type blueprint struct {
```

Comment:
@debasishg are you planning to allign this structure with the one I have in #129 ?
There is overlapping here.

Reply:
Same for the other structs.

Reply:
I looked at the `blueprint` structure in your PR. My requirement is a bit different and the struct I need is much smaller. Hence I decided to use a leaner version of the struct privately in the `deploy` package. But yes, some other structs will be reused.

Reply:
since verify will always happen as part of deploy, can't you use the same struct?

Reply:
will revisit this part once the verify code goes in ..